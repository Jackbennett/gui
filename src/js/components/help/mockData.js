export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
