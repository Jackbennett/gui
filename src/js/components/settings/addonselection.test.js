import React from 'react';
import { render } from '@testing-library/react';
import AddOnSelection from './addonselection';
import { undefineds } from '../../../../tests/mockData';

describe('AddOnSelection component', () => {
  it(`renders correctly`, () => {
    const { baseElement } = render(
      <AddOnSelection addOns={[{ name: 'configure', enabled: true }]} features={{ hasDeviceConfig: true, hasDeviceConnect: true }} />
    );
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
