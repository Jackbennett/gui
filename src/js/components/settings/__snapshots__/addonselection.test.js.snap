// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddOnSelection component renders correctly 1`] = `
<div
  class="flexbox column"
>
  <p>
    Extend Mender features with our add-ons. Select one or more from the list and submit to request add-ons to be added to your plan. We'll adjust your subscription and confirm it with you.
  </p>
  <div
    class="planPanel flexbox "
    style="align-items: center; margin-top: 10px;"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-4"
          data-indeterminate="false"
          type="checkbox"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </span>
    <div
      class="bold"
      style="min-width: 120px;"
    >
      Configure
    </div>
    <div
      class="flexbox column"
      style="min-width: 120px;"
    >
      <div
        class="link-color bold"
        style="font-size: 1rem;"
      >
        $10/month
      </div>
      <div>
        for first 50 devices
      </div>
    </div>
    <span
      class="info"
    >
      Expand your plan with device configuration features
    </span>
    <a
      class="margin-left-small"
      href="https://mender.io/plans/features"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn more
    </a>
  </div>
  <div
    class="planPanel flexbox "
    style="align-items: center; margin-top: 10px;"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-4"
          data-indeterminate="false"
          type="checkbox"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </span>
    <div
      class="bold"
      style="min-width: 120px;"
    >
      Troubleshoot
    </div>
    <div
      class="flexbox column"
      style="min-width: 120px;"
    >
      <div
        class="link-color bold"
        style="font-size: 1rem;"
      >
        $10/month
      </div>
      <div>
        for first 50 devices
      </div>
    </div>
    <span
      class="info"
    >
      Expand your plan with device troubleshooting features
    </span>
    <a
      class="margin-left-small"
      href="https://mender.io/plans/features"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn more
    </a>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained margin-top margin-bottom MuiButton-containedSecondary"
    style="align-self: flex-end; margin: 30px 0px;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Submit
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
